sklearn/linear_model/_ransac.py

```diff
@@ -357,9 +357,23 @@ def fit(self, X, y, sample_weight=None):
         if self.stop_probability < 0 or self.stop_probability > 1:
             raise ValueError("`stop_probability` must be in range [0, 1].")
 
+        estimator_fit_has_sample_weight = has_fit_parameter(estimator, "sample_weight")
+        estimator_name = type(estimator).__name__
+        if sample_weight is not None and not estimator_fit_has_sample_weight:
+            raise ValueError(
+                "%s does not support sample_weight. Samples"
```

Comment:
Can you use an f-string since we make a change here?

Reply:
updated